src/cursor.rs

```diff
@@ -293,45 +280,32 @@ impl Cursor {
         let cmd_name = cmd_type.to_str();
         let connstring = format!("{}", try!(socket.get_ref().peer_addr()));
 
-        let filter: bson::Document = match query.get("$query") {
+        let filter = match query.get("$query") {
             Some(&Bson::Document(ref doc)) => doc.clone(),
-            _ => doc!{},
+            _ => query.clone(),
         };
 
-        let projection = match return_field_selector {
-            Some(ref doc) => doc.clone(),
-            None => doc!{},
-        };
-
-        let sort = match query.get("$orderby") {
-            Some(&Bson::Document(ref doc)) => doc.clone(),
-            _ => doc!{},
-        };
 
         let command = match cmd_type {
             CommandType::Find => {
-                doc! {
-                "find" => coll_name,
-                "filter" => filter,
-                "projection" => projection,
-                "skip" => number_to_skip,
-                "limit" => number_to_return,
-                "batchSize" => batch_size,
-                "sort" => sort
-            }
+                let document = doc! { 
+                    "find" => coll_name,
+                    "filter" => filter
+                };
+
+                merge_options(document, options.clone())
             }
             _ => query.clone(),
         };
 
-
         let init_time = time::precise_time_ns();
         let result = Message::new_query(req_id,
                                         flags,
-                                        namespace.to_owned(),
-                                        number_to_skip,
-                                        batch_size,
-                                        query.clone(),
-                                        return_field_selector);
+                                        namespace.clone(),
+                                        options.skip.unwrap_or(0) as i32,
```

Comment:
I still have no idea why the spec says that the `skip` options should be an i64 if the wire protocol only accepts an i32. Oh, well...